### PR TITLE
Feat/tabs

### PR DESCRIPTION
**Pull request by danad02**
Created according to best practices - https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html.
With a little help from material-ui source - https://github.com/mui-org/material-ui/blob/next/packages/material-ui-lab/src/TabList/TabList.js and https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Tab/Tab.js

### CODE REVIEW

**Review by jankoritak — APPROVED** [review]
LGTM! Leaving just 2 nits.

File: packages/ui/src/Tabs/TabList.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * ### General Info
+ * - Expects direct children to have [role="tab"].
+ * - Implemented according to best practices: https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html.
+ *
+ */
+export const TabList: FC<TabsProps> = ({ ariaLabel, className, children }) => {
+  const tabListRef = useRef<HTMLDivElement>(null)
+
+  // Keyboard navigation assumes that the direct children have [role="tab"]
+  const handleKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
```
[the remark this answers is not in the document]
Comment (danad02) [reply packages/ui/src/Tabs/TabList.tsx:25]: Well, maybe we could at least wrap it with `useCallback`.

[the remark this answers is not in the document]
Comment (danad02) [reply packages/ui/src/Tabs/TabList.tsx:25]: Btw @jankoritak @utkukaratas @aigoncharov I was trying to test this function in the scope of the component but with no luck. I wasn't able to trigger focus on children with enzyme and just gave up after few hours - have you ever solve such problem?

File: packages/ui/src/Tabs/TabContext.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * A context used to pass valid values to Tab and TabPanel components.
+ */
+export const TabContext = createContext<TabContextValue>(tabContextDefaultValue)
+
+/**
+ * A custom hook utilizing context for easy usage.
+ */
+export const useTabContext = () => useContext(TabContext)
+
+export type TabContextComponentProps = { fullWidth?: boolean; children: ReactNode }
```
[the remark this answers is not in the document]
Comment (danad02) [reply packages/ui/src/Tabs/TabContext.tsx]: Changed here: https://github.com/hazelcast/frontend-shared/pull/171/commits/e0be52f2587ca45ac3094025704a1c2347c432d9

File: packages/ui/src/Tabs/TabList.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (danad02) [reply packages/ui/src/Tabs/TabList.tsx:25]: Update: I resolved issue with focus: https://github.com/hazelcast/frontend-shared/pull/171/commits/b01186c93b7699063a727488c5f707f5f53934b6

File: packages/ui/__tests__/Tabs/TabList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import { useUID } from 'react-uid'
+import { mountAndCheckA11Y } from '@hazelcast/test-helpers'
+// import { act } from 'react-dom/test-utils'
```
Comment (jankoritak) [inline packages/ui/__tests__/Tabs/TabList.test.tsx:4]:
```suggestion
```

File: packages/ui/__tests__/Tabs/TabPanel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  //   )
+
+  //   expect(wrapper.find('div').first().props()).toEqual({
+  //     className: styles.tabPanel,
+  //     role: 'tabpanel',
+  //     id: getPanelId(testId, index.toString()),
+  //     'aria-labelledby': getTabId(testId, index.toString()),
+  //     children: testChildren,
+  //   })
+  // })
```
Comment (jankoritak) [inline packages/ui/__tests__/Tabs/TabPanel.test.tsx:87]: Should remove this one?

```suggestion

```

**Review by jankoritak — APPROVED** [review]
The tests are looking super-exhaustive, well done!

File: packages/ui/__tests__/Tabs/TabList.test.tsx (hunk 3 above)
Comment (danad02) [reply packages/ui/__tests__/Tabs/TabList.test.tsx]: Removed https://github.com/hazelcast/frontend-shared/pull/171/commits/d99aa42502a57debaa685d1f53cd0f46027f71be

File: packages/ui/__tests__/Tabs/TabPanel.test.tsx (hunk 4 above)
Comment (danad02) [reply packages/ui/__tests__/Tabs/TabPanel.test.tsx]: Removed https://github.com/hazelcast/frontend-shared/pull/171/commits/e99d03b3b74b11df1c8ab34e8cae5d7ea606efdc

**Comment by danad02** [comment]
verify

**Pull request merged**